class-unlist-posts.php

```diff
@@ -63,6 +63,7 @@ public function init() {
 			add_filter( 'get_previous_post_where', array( $this, 'post_navigation_clause' ), 20, 1 );
 			add_action( 'wp_head', array( $this, 'hide_post_from_searchengines' ) );
 			add_filter( 'wp_robots', array( $this, 'no_robots_for_unlisted_posts' ) );
+			add_filter( 'rank_math/frontend/robots', array( $this, 'no_robots_for_unlisted_posts' ) );
```

Comment:
We can call a separate function here, for rankmath compatibility.
This function will check if current post is unlisted, if yes will return no robots.

This is so that we don't make the current function more complicated by introducing more conditions to it.

Reply:
Added a separate function `change_robots_for_rankmath` and check condtions in it.